examples/opensora_hpcai/opensora/schedulers/iddpm/gaussian_diffusion.py

```diff
@@ -432,18 +432,40 @@ def p_sample_loop_progressive(
     def ddim_sample(
         self,
         model,
-        x,
+        x: Tensor,
         t,
         clip_denoised=True,
         denoised_fn=None,
         cond_fn=None,
         model_kwargs=None,
         eta=0.0,
+        frames_mask: Optional[np.ndarray] = None,
     ):
         """
         Sample x_{t-1} from the model using DDIM.
         Same usage as p_sample().
         """
+        if frames_mask is not None:
+            if frames_mask.shape[0] != x.shape[0]:
+                frames_mask = frames_mask.reshape(1, -1).repeat(2, axis=0)  # HACK
```

Comment:
is it verified on CANN0705 + MS0705? may need to cast frames_mask to int32 for repeat ops for better performance